### PR TITLE
Add DAQmx triggering and shared timebase example

### PR DESCRIPTION
**Pull request by gregstoll**
### What does this Pull Request accomplish?

Adds a triggering and shared timebase example. I manually converted this from the C example, although the callback parts of the code are fairly different than in C.

### Why should this Pull Request be merged?

This is a relatively complex example that involves asyncio, triggering, and sharing a timebase.

### What testing has been done?

I can't run the full example because I don't have the necessary hardware, but I was able to comment out all the follower and synchronization code and verify that data gets acquired and pressing Enter or Ctrl+C stops the example.

### CODE REVIEW

File: examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+import sys
+
+import nidaqmx_pb2 as nidaqmx_types
+import nidaqmx_pb2_grpc as grpc_nidaqmx
+
+# False = sample clock, True = reference clock
+synchronization_type_is_reference_clock = False
+server_address = "localhost"
+server_port = "31763"
+leader_device = "Dev1"
```
Comment (astarche) [inline examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py:98]: Nice terminology fix!

File: examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+#       well as and acquisition start trigger and generation start
+#       trigger on the follower. The signal is shared along the PXI_Trig
+#       / RTSI bus.
+#    6. Synthesize a standard sine waveform and load this data into
+#       the output RAM buffer for each device.
+#    7. Call the Start function to start the acquisition.
+#    8. Read all of the data continuously. The 'Samples per Channel'
+#       control will specify how many samples per channel are read
+#       each time. If either device reports an error or the user
+#       presses Enter, the acquisition will stop.
```
Comment (astarche) [inline examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py:53]: PR Comment says `Ctrl+C` not Enter. but the code does look like Enter.

File: examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
+# This example configures a total of four tasks on two PXI Dynamic Signal Acquistion (DSA) devices.
+# Each device runs a task for analog input and another for analog output. This example illustrates
+# continuous analog input and output operations.
+#
+# Instructions for Running:
+#    1. Select which type of Synchronization method you want to use.
+#    2. Select the physical channel to correspond to where your
```
Comment (astarche) [inline examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py:7]: "to correspond to where your signal is input" -> "corresponding to the input signal"?

From whence it corresponds?

I assume this is copied from the C API doc, so I'm OK with it as is. (And the rest of it is phrased really well. Just this line got overcooked).

**Review by astarche — APPROVED** [review]

File: examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py (hunk 2 above)
Comment (gregstoll) [reply examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py:53]: It actually quits in either case. I want to document it as Enter, but I also wanted to make sure that `Ctrl+C` works to quit (which it didn't before I did some stuff)

File: examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py (hunk 3 above)
Comment (gregstoll) [reply examples/nidaqmx/dsa-shared-timebase-and-trig-analog-input-and-output.py]: Yeah, I like your suggestion. (and yeah, I did copy the docs over, but I did fix a few typos along the way, so might as well fix this too!)

**Review by harsha-bhushan — APPROVED** [review]

**Pull request merged**